lib/Dialect/Calyx/Transforms/CalyxLoweringUtils.cpp

```diff
@@ -191,13 +199,27 @@ Value MemoryInterface::writeEn() {
   return std::get<MemoryPortsImpl>(impl).writeEn;
 }
 
+std::optional<Value> MemoryInterface::readEn() {
+  if (auto *memOp = std::get_if<calyx::MemoryOp>(&impl); memOp) {
+    return nullptr;
+  }
```

Comment:
Remove braces. Since you're not using `memOp` shouldn't `if (std::get_if<calyx::MemoryOp>(&impl))` suffice? And if that's the case, `std::holds_alternative` may suffice.